run.py

```diff
@@ -28,70 +43,110 @@ def get_config(cfg):
         sys.exit()
 
+
 def check_test_exists(directory, test_name):
-    """
-    Checks if mentioned tests exist in particular directory or not
-    """
-    directory_path = os.path.join(os.getcwd(), 'tests') #name of directory under which test exists
-    if(test_name!= "statFile"):
+    # Checks if mentioned tests exist in particular directory or not
+
+    directory_path = os.path.join(os.getcwd(), 'tests')  # name of directory under which test exists
+    if test_name != "statFile":
         test_file = "test_" + test_name + ".py"
         test_file_path = os.path.join(os.path.join(directory_path, directory), test_file)
         if not os.path.exists(test_file_path):
-            raise Exception( test_name + " test file does not exists")
+            raise Exception(test_name + " test file does not exists")
+
 
 def check_input_validity(params):
-    """
-    Check the parameter type
-    """
-    string_val = ['filePath', 'fileSize','repoName', 'repoSize']
+    # Checks the parameter type
+
+    string_val = ['filePath', 'fileSize', 'repoName', 'repoSize']
     int_val = ['fileNumber', 'num']
     for key, value in params.items():
         if key in int_val:
             if not type(value) == int:
-                raise Exception(key + " having value " + str(value)+ " is not integer")
+                raise Exception(key + " having value %s is not a integer" % str(value))
         if key in string_val:
-            if not type(value)==str:
-                raise Exception( key + " having value" + str(value) + " is not string")
-
-
+            if not type(value) == str:
+                raise Exception(key + " having value %s is not a string" % str(value))
 
 
 def validator(tasks):
-    """
-    To check validity of YAML File
-    """
+    # To check validity of YAML File
+
     for test in tasks.values():
-        for directory, component_test in test.items(): #storage
-            if(component_test!=None):
+        for directory, component_test in test.items():  # storage
+            if component_test:
                 for test_name, param in component_test.items():
-                    if(param!=None):
+                    if param:
                         check_test_exists(directory, test_name)
                         check_input_validity(param)
 
 
 def cleanup():
-    """
-    delete the created files
-    """
+    # delete the created files
+
     filelist = glob.glob(os.path.join("", "*.txt"))
     print(filelist)
     for f in filelist:
         os.remove(f)
 
+
+def docker_exec(container_name, arg, user=None, working_dir=None):
+    run_script = "python3 run_container.py --test"
+    if working_dir and user:
+        cmd = "sudo docker exec -it -u %s -w %s %s %s %s" % (user, working_dir, container_name, run_script, arg)
+    else:
+        cmd = "sudo docker exec -it %s %s %s" % (container_name, run_script, arg)
+    os.system(cmd)
+
+
+def docker_cp_from_container(container_name, path, user=None):
```

Comment:
Shouldn't this be part of the helper functions?

Reply:
Done